test/integration/smoke/test_outofbandmanagement.py

```diff
@@ -0,0 +1,561 @@
+        )[0].value
+
+        self.configureAndEnableOobm()
+        self.startIpmiServer()
+        bmc = IpmiServerContext().bmc
+
+        bmc.powerstate = 'on'
+        self.checkSyncToState('On', interval)
+
+        bmc.powerstate = 'off'
+        self.checkSyncToState('Off', interval)
+
+        self.server.shutdown()
+        self.server.server_close()
+
+        # Check for unknown state (ipmi server not reachable)
+        self.checkSyncToState('Unknown', interval)
+
+
+    @attr(tags=["smoke", "advanced"])
+    def test_oobm_multiple_mgmt_server_ownership(self):
+        """
+            Tests out-of-band management ownership expiry across multi-mgmt server
+        """
+        self.configureAndEnableOobm()
+
+        cloudstackVersion = Configurations.listCapabilities(self.apiclient).cloudstackversion
+
+        currentMsHosts = []
+        mshosts = self.dbclient.execute("select msid from mshost where version='%s' and removed is NULL and state='Up'" % (cloudstackVersion))
+        if len(mshosts) > 0:
+            currentMsHosts = map(lambda row: row[0], mshosts)
+
+        # Inject fake ms host
+        self.dbclient.execute("insert into mshost (msid,runid,name,state,version,service_ip,service_port,last_update) values (%s,%s,'oobm-marvin-fakebox', 'Down', '%s', '127.0.0.1', '22', NOW())" % (self.getFakeMsId(), self.getFakeMsRunId(), cloudstackVersion))
+
+        # Pass ownership to the fake ms id
+        self.dbclient.execute("update oobm set mgmt_server_id=%d where port=%d" % (self.getFakeMsId(), self.getIpmiServerPort()))
+
+        self.debug("Testing oobm background sync")
+        pingInterval = float(list_configurations(
+            self.apiclient,
+            name='ping.interval'
+        )[0].value)
+
+        pingTimeout = float(list_configurations(
+            self.apiclient,
+            name='ping.timeout'
+        )[0].value)
+
+        # Sleep until the ownership gets expired as the fake ms id is not reachable
+        for _ in range(10):
+            rows = self.dbclient.execute("select * from mshost_peer where peer_runid=%s" % self.getFakeMsRunId())
+            if len(rows) > 0:
+                self.debug("Mgmt server is now trying to contact the fake mgmt server")
+                self.dbclient.execute("update mshost set removed=now() where runid=%s" % self.getFakeMsRunId())
+                self.dbclient.execute("update mshost_peer set peer_state='Down' where peer_runid=%s" % self.getFakeMsRunId())
+                break
+            time.sleep(1 + (pingInterval * pingTimeout / 10))
+
+        for _ in range(100):
+            rows = self.dbclient.execute("select mgmt_server_id from oobm where port=%d" % (self.getIpmiServerPort()))
+            if len(rows) > 0 and rows[0][0] != self.getFakeMsId():
+                self.debug("Out-of-band management ownership expired as node was detected to be gone")
+                break
+            time.sleep(1 + (pingInterval * pingTimeout / 10))
```

Comment:
Consider using the ``wait_until`` function to collapse both of these wait loops.